packages/@aws-cdk/aws-eks/lib/fargate-profile.ts

```diff
@@ -0,0 +1,192 @@
+   * all of the labels that are specified in the selector for it to be
+   * considered a match.
+   *
+   * @default - all pods within the namespace will be selected.
+   */
+  readonly labels?: { [key: string]: string };
+}
+
+/**
+ * Fargate profiles allows an administrator to declare which pods run on
+ * Fargate. This declaration is done through the profile’s selectors. Each
+ * profile can have up to five selectors that contain a namespace and optional
+ * labels. You must define a namespace for every selector. The label field
+ * consists of multiple optional key-value pairs. Pods that match a selector (by
+ * matching a namespace for the selector and all of the labels specified in the
+ * selector) are scheduled on Fargate. If a namespace selector is defined
+ * without any labels, Amazon EKS will attempt to schedule all pods that run in
+ * that namespace onto Fargate using the profile. If a to-be-scheduled pod
+ * matches any of the selectors in the Fargate profile, then that pod is
+ * scheduled on Fargate.
+ *
+ * If a pod matches multiple Fargate profiles, Amazon EKS picks one of the
+ * matches at random. In this case, you can specify which profile a pod should
+ * use by adding the following Kubernetes label to the pod specification:
+ * eks.amazonaws.com/fargate-profile: profile_name. However, the pod must still
+ * match a selector in that profile in order to be scheduled onto Fargate.
+ */
+export class FargateProfile extends Construct implements ITaggable {
+
+  /**
+   * The full Amazon Resource Name (ARN) of the Fargate profile.
+   *
+   * @attribute
+   */
+  public readonly fargateProfileArn: string;
+
+  /**
+   * The name of the Fargate profile.
+   *
+   * @attribute
+   */
+  public readonly fargateProfileName: string;
+
+  /**
+   * Resource tags.
+   */
+  public readonly tags: TagManager;
+
+  constructor(scope: Construct, id: string, props: FargateProfileProps) {
+    super(scope, id);
+
+    const provider = ClusterResourceProvider.getOrCreate(this);
+
+    const role = props.podExecutionRole ?? new iam.Role(this, 'PodExecutionRole', {
+      assumedBy: new iam.ServicePrincipal('eks-fargate-pods.amazonaws.com'),
+      managedPolicies: [ iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonEKSFargatePodExecutionRolePolicy') ]
+    });
+
+    let subnets: string[] | undefined;
+    if (props.vpc) {
+      const selection: ec2.SubnetSelection = props.subnetSelection ?? { subnetType: ec2.SubnetType.PRIVATE };
+      subnets = props.vpc.selectSubnets(selection).subnetIds;
+    }
+
+    if (props.selectors.length < 1) {
+      throw new Error(`Fargate profile requires at least one selector`);
+    }
+
+    if (props.selectors.length > 5) {
+      throw new Error(`Fargate profile supports up to five selectors`);
+    }
+
+    this.tags = new TagManager(TagType.MAP, 'AWS::EKS::FargateProfile');
+    for (const [ key, value ] of Object.entries(props.tags || {})) {
+      this.tags.setTag(key, value);
```

Comment:
Why not leave this up to the aspect-based tagger?

Reply:
Are you proposing not to expose `tags` as a property?

Reply:
Yes?

Reply:
OK. I'll remove. Seems like that's the common pattern (we should awslint this shit).